src/wallet/db.cpp

```diff
@@ -390,8 +390,11 @@ CDB::CDB(CWalletDBWrapper& dbw, const char* pszMode, bool fFlushOnCloseIn) : pdb
             if (fMockDb) {
                 DbMpoolFile* mpf = pdb->get_mpf();
                 ret = mpf->set_flags(DB_MPOOL_NOFILE, 1);
-                if (ret != 0)
+                if (ret != 0) {
+                    delete pdb;
```

Comment:
Maybe another way you could fix this without adding a delete call would be to declare pdb as a unique_ptr in the header? This would also let you remove the other deletes below.

Reply:
And in `std::map<std::string, Db*> CDBEnv::mapDb`?

Reply:
> And in std::map\<std::string, Db*> CDBEnv::mapDb?

Oops, sorry. The pdb member should _not_ be a unique_ptr because the pointer is actually owned (deleted) by mapdb, not the cdb class.

If you wanted to do a minimal fix you could declare a unique_ptr\<Db> temp_db here and then do pdb = temp_db.release() at the end of this function.

If you wanted to go further, you could change mapDb to hold unique_ptrs and std::move from temp_db into the map at the end of this function.

Reply:
I wouldn't go that further in a bugfix. But I'll probably do that in a different PR.

Reply:
Ok, 0d3ec0c3a15e7afde42b3b98e369d989a6d9e7f7 is the minimal fix for reference.

Reply:
Anyway, the call `Exists` below uses `pdb` internally. I believe this is the minimal fix.

Reply:
> Ok, 0d3ec0c is the minimal fix for reference.

Had the same, but see my previous comment.

Reply:
Also, I thought of not doing `pdb = nullptr` but if this code is moved to some member function then that can a problem.

Reply:
Well, I can `.release()` before `Exists()`.